Classes/Controller/BackendAdminController.php

```diff
@@ -0,0 +1,355 @@
+                } else {
+                    throw new \Exception('Missing client configuration: "owner id"');
+                }
+
+                $this->documentRepository->update($document);
+
+                // Move files to the target client
+                $files = $document->getFile();
+                foreach ($files as $file) {
+                    $file->setPid($client->getPid());
+                    $this->fileRepository->update($file);
+                }
+
+                // If there are bookmarks, they have to be adjusted, depending on the existence of the related user
+                // inside the target client, and either deleted or moved to the target client.
+                $this->bookmarkRepository->crossClient(true);
+                $bookmarks = $this->bookmarkRepository->findDocumentBookmarks($document);
+                /** @var Bookmark $bookmark */
+                foreach ($bookmarks as $bookmark) {
+                    if ($this->frontendUserRepository->isUserInClient($bookmark->getFeUserUid(), $client->getPid())) {
+                        $bookmark->setPid($client->getPid());
+                        $this->bookmarkRepository->update($bookmark);
+                    } else {
+                        $this->bookmarkRepository->remove($bookmark);
+                    }
+                }
+
+                // Move document into the target search index.
+                $elasticSearch = new ElasticSearch($currentClient->getPid());
+                $elasticSearch->delete($document->getDocumentIdentifier());
+                $targetElasticSearch = new ElasticSearch($client->getPid());
+                $targetElasticSearch->index($document);
+
+                $this->flashMessage(
+                   'client_changed',
+                   'client_changed_message',
+                   AbstractMessage::OK,
+                   [],
+                   [$identifier]
+                );
+           } else {
+               $this->flashMessage(
+                   'change_client_forbidden',
+                   'change_client_forbidden_message',
+                   AbstractMessage::ERROR,
+                   [],
+                   [$identifier]
+               );
+           }
+
+           $this->view->assign('currentClient', $client);
+           $this->view->assign('document', $document);
+       } else {
+            $this->flashMessage(
+                'missing_document_type',
+                'missing_document_type_message',
+                AbstractMessage::ERROR,
+                [],
+                [$identifier]
+            );
+
+            $this->forward(
+              'chooseNewClient',
+              null,
+              null,
+              [
+                  'document' => $document,
+                  'identifier' => $identifier
+              ]
+            );
+       }
+    }
+
+    /**
+     * @param Document $document
+     */
+    protected function canChangeClient(Document $document)
```

Comment:
This function might be a better fit for the Document class. Or a dedicated Workflow class if it exists. Because why should this web controller have business logic after all? It's non-critical for now.